db/utils.js

```diff
@@ -10,4 +10,23 @@ exports.checkArticleExists = (article_id) => {
         return Promise.reject({status:404,msg:'article not found'})
         }
     })
-}
\ No newline at end of file
+}
+
+async function selectCommentsOfArticle(article_id){
+    return db.query(`SELECT * FROM comments WHERE article_id = $1;`,[article_id])
+}
+async function addCommentsCountColumn(){
+    return db.query(`ALTER TABLE articles ADD comments_count INT DEFAULT 0;`)
+}
+
+exports.addCommentsCountToArticle = async (article_id) =>{
+    const alterResult = await addCommentsCountColumn();
+    const comments = await selectCommentsOfArticle(article_id);
+    const comments_count = comments.rows.length;
+    return db.query(`UPDATE articles SET comments_count = ${comments_count} WHERE article_id = $1 RETURNING*;`
+        ,[article_id])
+    .then((updatedArticle) => {
+        return updatedArticle.rows[0]
+    })
+}
+
```

Comment:
Similarly to the feedback given on your #11 ticket, you don't actually need these util functions.

Comment count can be obtained using a JOIN on the comments table then using the COUNT keyword in psql to count the number of comments on each article.

You don't need to add a comments column onto the articles table - comments is it's own table with comments that refer to the article they belong to with foreign keys. To achieve something similar to what you're trying to do with this util function - we join the tables during our sql queries.

This means you can do this ticket in a simpler way without any of these util functions